modules/solutions/twitter-analytics/siddhi-extensions/siddhi-execution-country/component/src/main/java/org/wso2/extension/siddhi/execution/country/CountryFunction.java

```diff
@@ -0,0 +1,179 @@
+        examples = @Example(description = "This outputs the country of the provided location. In this scenario, the " +
+                "output is 'India' .", syntax = "country(\"New Delhi, India\")")
+)
+
+public class CountryFunction extends FunctionExecutor {
+
+    private static final Logger log = Logger.getLogger(CountryFunction.class);
+
+    private Attribute.Type returnType = Attribute.Type.STRING;
+
+    /**
+     * The initialization method for {@link FunctionExecutor}, which will be called before other methods and validate
+     * the all configuration and getting the initial values.
+     *
+     * @param attributeExpressionExecutors are the executors of each attributes in the Function
+     * @param configReader                 this hold the {@link FunctionExecutor} extensions configuration reader.
+     * @param siddhiAppContext             Siddhi app runtime context
+     */
+    @Override
+    protected void init(ExpressionExecutor[] attributeExpressionExecutors, ConfigReader configReader,
+                        SiddhiAppContext siddhiAppContext) {
+        if (attributeExpressionExecutors.length != 1) {
+            throw new SiddhiAppValidationException("Invalid no of arguments passed to find:country() function. " +
+                    "Required 1. Found " + attributeExpressionExecutors.length);
+        } else if (attributeExpressionExecutors[0].getReturnType() != Attribute.Type.STRING) {
+            throw new SiddhiAppValidationException(
+                    "Invalid parameter type found for find:country() function, required " + Attribute.Type.STRING +
+                            ", " + "but found " + attributeExpressionExecutors[0].getReturnType());
+        }
+    }
+
+    /**
+     * The main execution method which will be called upon event arrival
+     * when there are more than one Function parameter
+     *
+     * @param data the runtime values of Function parameters
+     * @return the Function result
+     */
+    @Override
+    protected Object execute(Object[] data) {
+        return null; //Since the length function takes in only 1 parameter, this method does not get called.
+        // Hence, not implemented.
+    }
+
+    /**
+     * The main execution method which will be called upon event arrival
+     * when there are zero or one Function parameter
+     *
+     * @param data null if the Function parameter count is zero or
+     *             runtime data value of the Function parameter
+     * @return the Function result
+     */
+    @Override
+    protected Object execute(Object data) {
+        String[] location;
+        String[] countryValues;
+        String line;
+        int i;
+        int j;
+        if (data == null) {
+            throw new SiddhiAppRuntimeException("Invalid input given to find:length() function. " +
+                    "The argument cannot be null");
+        }
+        InputStream inputStream = CountryFunction.class.getResourceAsStream("/Countries.csv");
```

Comment:
Why are you loading this `Countries.csv` per each event? Why is it not possible to load it at the extension init?